openhands/server/listen_socket.py

```diff
@@ -132,20 +132,21 @@ async def connect(connection_id: str, environ: dict) -> None:
         agent_state_changed = None
         if agent_loop_info is None:
             raise ConnectionRefusedError('Failed to join conversation')
-        async_store = AsyncEventStoreWrapper(
-            agent_loop_info.event_store, latest_event_id + 1
-        )
-        async for event in async_store:
-            logger.debug(f'oh_event: {event.__class__.__name__}')
-            if isinstance(
-                event,
-                (NullAction, NullObservation, RecallAction),
-            ):
-                continue
-            elif isinstance(event, AgentStateChangedObservation):
-                agent_state_changed = event
-            else:
-                await sio.emit('oh_event', event_to_dict(event), to=connection_id)
+        if agent_loop_info.event_store:
```

Comment:
This will silently do nothing if `agent_loop_info.event_store` is `None`, is that the proper behavior? Or is the proper behavior to throw an error?

Reply:
That's a good question, I don't know enough about the scenario it could be None in. My thinking was "Well we're looking over all the events so if there's no store there are definitely no events to loop over".

But I can equally argue that AsyncEventStoreWrapper would have exploded before so we should preserve the behavior unless we know we want different.